linalg/sparsemat.cpp

```diff
@@ -650,18 +653,23 @@ void SparseMatrix::AddMultTranspose(const Vector &x, Vector &y,
    }
    else
    {
-      MFEM_VERIFY(Device::IsDisabled(), "transpose action on device is not "
-                  "enabled; see BuildTranspose() for details.");
-      for (int i = 0; i < height; i++)
+      const int height = this->height;
+      const int nnz = J.Capacity();
+      auto d_I = Read(I, height+1);
+      auto d_J = Read(J, nnz);
+      auto d_A = Read(A, nnz);
+      auto d_x = x.Read();
+      auto d_y = y.ReadWrite();
+      MFEM_FORALL(i, height,
```

Comment:
This loop requires atomics -- that's why we kept it on host and constructed the transpose explicitly.

We should probably revert the changes here.

Reply:
I reverted the `Device` for loop and forced it on `Host`, we have some tests that go through this path.